orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java

```diff
@@ -0,0 +1,131 @@
+        TableProperties.ORC_STRIPE_SIZE_BYTES, String.valueOf(stripeSizeBytes),
+        TableProperties.ORC_BLOCK_SIZE_BYTES, String.valueOf(blockSizeBytes),
+        TableProperties.DEFAULT_FILE_FORMAT, FileFormat.ORC.name());
+
+    File folder = TEMPORARY_FOLDER.newFolder();
+
+    String warehouse = folder.getAbsolutePath();
+    String tablePath = warehouse.concat("/test");
+    Assert.assertTrue("Should create the table path correctly.", new File(tablePath).mkdir());
+
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Table table = new HadoopTables().create(SCHEMA, spec, properties, tablePath);
+
+    File testFile = TEMPORARY_FOLDER.newFile();
+    Assert.assertTrue(testFile.delete());
+
+    FileAppender<Record> writer = ORC.write(Files.localOutput(testFile))
+        .forTable(table)
+        .createWriterFunc(GenericOrcWriter::buildWriter)
+        .build();
+
+    DynFields.BoundField<Configuration> confField =
+        DynFields.builder().hiddenImpl(writer.getClass(), "conf").build(writer);
+
+    Configuration configuration = confField.get();
+    Assert.assertEquals(String.valueOf(blockSizeBytes), configuration.get(OrcConf.BLOCK_SIZE.getAttribute()));
+    Assert.assertEquals(String.valueOf(stripeSizeBytes), configuration.get(OrcConf.STRIPE_SIZE.getAttribute()));
+    Assert.assertEquals(FileFormat.ORC.name(), configuration.get(TableProperties.DEFAULT_FILE_FORMAT));
+  }
+
+  @Test
+  public void testOrcTableDeleteProperties() throws Exception {
+    Long stripeSizeBytes = 32L * 1024 * 1024;
+    Long blockSizeBytes = 128L * 1024 * 1024;
+
+    ImmutableMap<String, String> properties = ImmutableMap.of(
+        TableProperties.DELETE_ORC_STRIPE_SIZE_BYTES, String.valueOf(stripeSizeBytes),
+        TableProperties.DELETE_ORC_BLOCK_SIZE_BYTES, String.valueOf(blockSizeBytes),
+        TableProperties.DEFAULT_FILE_FORMAT, FileFormat.ORC.name());
+
+    File folder = TEMPORARY_FOLDER.newFolder();
+
+    String warehouse = folder.getAbsolutePath();
+    String tablePath = warehouse.concat("/test");
+    Assert.assertTrue("Should create the table path correctly.", new File(tablePath).mkdir());
+
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Table table = new HadoopTables().create(SCHEMA, spec, properties, tablePath);
+
+    File testFile = TEMPORARY_FOLDER.newFile();
+    Assert.assertTrue(testFile.delete());
+
+    EqualityDeleteWriter<Object> deleteWriter = ORC.writeDeletes(Files.localOutput(testFile))
+        .forTable(table)
+        .equalityFieldIds(1)
+        .createWriterFunc(GenericOrcWriter::buildWriter)
+        .buildEqualityWriter();
+
+
+    DynFields.BoundField<OrcFileAppender> writer =
+        DynFields.builder().hiddenImpl(deleteWriter.getClass(), "appender").build(deleteWriter);
+
+    OrcFileAppender orcFileAppender = writer.get();
+    DynFields.BoundField<Configuration> confField =
+        DynFields.builder().hiddenImpl(orcFileAppender.getClass(), "conf").build(orcFileAppender);
```

Comment:
Nit: Please use the `OrcFileAppender<Record>` here to eliminate the idea warnings.